tests/test-naughty-legacy.lua

```diff
@@ -698,10 +742,9 @@ table.insert(steps, function()
     assert(n.text    == "baz")
 
     -- Test the ID system
-    n.id = 1337
-    assert(n.id == 1337)
-    assert(naughty.getById(1337) == n)
-    assert(naughty.get_by_id(1337) == n)
+    local id = n.id
+    assert(naughty.getById(id) == n)
```

Comment:
is that camelcase notation supported for all the methods as an alternative to underscore?!

Reply:
more like a snake case alternative to an API that used CamelCase. It's not the only example. There is about a hundred API elements using nocase or CamelCase that predates the standardization to snake_case. Some are still not deprecated (mostly in awful).

Reply:
but i meant are those both methods are implemented explicitly or we have some general handler for that?

Reply:
> but i meant are those both methods are implemented explicitly or we have some general handler for that?

All existing CamelCase and nocase functions and method part of the public API get deprecated when I work in the module they belong to. The CamelCase and nocase end up with a `gears.deprecate` call and then call the snake_case equivalent.